File: app/src/main/java/com/example/owner_pc/animechecker/contract/AnimeDetailContract.java
package com.example.owner_pc.animechecker.contract;

import com.example.owner_pc.animechecker.model.entity.Anime;
import com.example.owner_pc.animechecker.model.entity.AnimePage;

import java.util.List;

/**
 * Created by owner-PC on 2017/05/22.
 */

public interface AnimeDetailContract {

    AnimePage getAnimePage();

    void startBrowser(String url);

    //// TODO: 2017/05/29 AnimeCardではなくAnimeで渡した場合に変更する
    void showDirectorAnimes(List<Anime> animes);

    void showStudioAnimes(List<Anime> animes);

    void showError(String message);
}

File: app/src/main/java/com/example/owner_pc/animechecker/contract/AnimeDetailContract.kt
package com.example.owner_pc.animechecker.contract

import com.example.owner_pc.animechecker.model.entity.Anime
import com.example.owner_pc.animechecker.model.entity.AnimePage

/**
 * Created by owner-PC on 2017/05/22.
 */

interface AnimeDetailContract {

    val animePage: AnimePage

    fun startBrowser(url: String)

    //// TODO: 2017/05/29 AnimeCardではなくAnimeで渡した場合に変更する
    fun showDirectorAnimes(animes: List<Anime>)

    fun showStudioAnimes(animes: List<Anime>)

    fun showError(message: String)
}
